### PR TITLE
live-tests: implement debug mode

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/2858
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/2853

This PR introduces a usable `debug` command under the new `live-tests` CLI.
It also adds strong re-usable primitives that we will re-use in an upcoming `regression-test` command (which will use pytest).

## How
```bash
live-tests debug read \
--connector-image=airbyte/source-pokeapi:dev \
--connector-image=airbyte/source-pokeapi:latest \
--config-path=poke_config.json \
--catalog-path=configured_catalog.json
```

This command will run the selected connector command on the selected connector images and will dump the command output to files for further manual analysis.

An HTTP proxy is introduced to cache requests on the same connectors.

More details in the [`README.md`](https://github.com/airbytehq/airbyte/blob/cf7cdb73dfc37a5588113dc97ed1cf5019be3fe6/airbyte-ci/connectors/live-tests/README.md)

### CODE REVIEW

**Comment by alafanechere** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35786?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#35786** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35786?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#35624** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35624?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/live-tests/src/live_tests/commons/models.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        stdout_file_path = final_dir / self.stdout_filename
+        stdout_file_path.write_text(self.execution_result.stdout)
+
+        stderr_file_path = final_dir / self.stderr_filename
+        stderr_file_path.write_text(self.execution_result.stderr)
+        if self.execution_result.http_dump:
+            http_dump_file_path = final_dir / self.http_dump_filename
+            await self.execution_result.http_dump.export(str(http_dump_file_path.resolve()))
+        # TODO merge ExecutionReport.save_to_disk and Backend.write?
+        # Make backends use customizable
```
Comment (clnoll) [inline airbyte-ci/connectors/live-tests/src/live_tests/commons/models.py:232]: I'm curious to know what you have in mind here.

File: airbyte-ci/connectors/live-tests/src/live_tests/commons/connector_runner.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            container = container.with_new_file(self.IN_CONTAINER_CONFIG_PATH, contents=json.dumps(dict(self.config)))
+        if self.state:
+            container = container.with_new_file(self.IN_CONTAINER_STATE_PATH, contents=json.dumps(self.state))
+        if self.catalog:
+            container = container.with_new_file(self.IN_CONTAINER_CATALOG_PATH, contents=self.catalog.json())
+        if self.enable_http_cache:
+            container = await self.bind_connector_container_to_proxy(container)
+        executed_container = await container.with_exec(self.full_command).sync()
+
+        return ExecutionResult(
```
Comment (clnoll) [inline airbyte-ci/connectors/live-tests/src/live_tests/commons/connector_runner.py:212]: I like the new pattern of `ExecutionInput`/`ExecutionResult`/`ExecutionReport`.

**Review by clnoll — APPROVED** [review]
Looks great! Just a few minor comments & questions, no blockers. Excited to get the first version out!

File: airbyte-ci/connectors/live-tests/src/live_tests/commons/models.py (hunk 1 above)
Comment (alafanechere) [reply airbyte-ci/connectors/live-tests/src/live_tests/commons/models.py:232]: I basically  realized that to classes perform final writes to file system of artifact, could be better to have this centralized.

**Pull request merged**